cmd/tailscaled/tailscaled_windows.go

```diff
@@ -184,6 +191,17 @@ func (service *ipnService) Execute(args []string, r <-chan svc.ChangeRequest, ch
 				syslogf("Service session change notification")
 				handleSessionChange(cmd)
 				changes <- cmd.CurrentStatus
+			case cmdUninstallWinTun:
+				syslogf("Stopping tailscaled child process and uninstalling WinTun")
+				childDoneCh := doneCh
+				doneCh = make(chan struct{})
```

Comment:
This channel seems unused, except the close in the go routine below

Reply:
I think this needs some more comments to explain the flow, but from what I can tell this is what's happening:
1. The existing `doneCh` that will be closed when the subprocess exits is saved in `childDoneCh`.
2. `doneCh` is replaced with a new channel so this event loop will continue even after the subprocess is gone.
3. `cancel()` triggers shutting down the subprocess
4. A new goroutine finishes the wintun uninstall process, the current goroutine goes back to the service event loop.
5. The new goroutine waits for the subprocess to finish exiting, then it uninstalls wintun, followed by closing `doneCh` so the service event loop exits.

I think this looks correct, although maybe there should be a service status update indicating that it's in the process of shutting down?

Reply:
Will add comments. I was a bit iffy on whether to change the status to StopPending when receiving the event, but I think you've convinced me.